pyuploadcare/api_resources.py

```diff
@@ -624,52 +616,52 @@ class BaseApiList(object):
     constructor = None
     filters = []
 
-    def __init__(self, **kwargs):
-        # Only explicit kwargs are allowed.
-
-        if kwargs.get('since') is not None and kwargs.get('until') is not None:
-            raise TypeError('Only one of since and until arguments is allowed.')
-
-        self.since = kwargs.pop('since', None)
-        self.until = kwargs.pop('until', None)
-        self.limit = kwargs.pop('limit', None)
-        self.request_limit = kwargs.pop('request_limit', None)
+    def __init__(self, starting_point=None, ordering=None, limit=None,
+                 request_limit=None, stored=None, removed=None):
+        self.starting_point = starting_point
+        self.ordering = ordering
+        self.limit = limit
+        self.request_limit = request_limit
+        self.stored = stored
+        self.removed = removed
         self._count = None
 
-        for f, default in self.filters:
-            setattr(self, f, kwargs.pop(f, default))
-
-        if kwargs:
-            raise TypeError('Extra arguments: {}.'.format(
-                ", ".join(kwargs.keys())
-            ))
+        # Makes sure that the value of starting_point a correctly formatted
+        ordering_field = (ordering or '').lstrip('-')
+        if ordering_field in ('', 'datetime_uploaded') and starting_point:
+            self.starting_point = dateutil.parser.parse(
```

Comment:
I disagree that we need this in core API. At least, this disallows `datetime` type.